darwin/future/tests/core/items/test_set_priority.py

```diff
@@ -0,0 +1,88 @@
+from unittest.mock import Mock
+from uuid import UUID
+
+import pytest
+import responses
+
+from darwin.exceptions import DarwinException
+from darwin.future.core.client import ClientCore
+from darwin.future.core.items.set_item_priority import set_item_priority
+from darwin.future.tests.core.fixtures import *
+
+
+@responses.activate
+def test_set_item_priority(base_client) -> None:
+    responses.add(
+        responses.POST,
+        base_client.config.api_endpoint + "v2/teams/test-team/items/priority",
+        json={"affected_item_count": 0},
+    )
+
+    response = set_item_priority(
+        base_client,
+        "test-team",
+        123,
+        UUID("00000000-0000-0000-0000-000000000000"),
+        999,
+    )
+
+    assert response == {"affected_item_count": 0}
+
+
+def test_set_item_priority_with_filters() -> None:
+    base_client = Mock(spec=ClientCore)
+
+    expected_payload = {
+        "priority": 10,
+        "filters": {
+            "dataset_ids": [123],
+            "item_ids": ["00000000-0000-0000-0000-000000000000"],
+            "status": "open",
+        },
+    }
+
+    # Define the expected endpoint
+    expected_endpoint = "/v2/teams/test-team/items/priority"
+
+    # Define the expected response
+    expected_response = {"status": "success"}
```

Comment:
I might be missing something, but if successful doesn't the endpoint always return `{"affected_item_count": n}`?

I couldn't get it to return `{"status": "success"}`

Reply:
If you pass it a 422 inducing load of rubbish, it returns a response including `errors`

`Graphotate.Common.APIError` example:
```
{
  "errors": {
    "code": "EXTERNAL_STORAGE_LIMIT_REACHED",
    "detail": {},
    "message": "Error message",
    "status": 422
  }
}```